File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n=len(prices)
        profit=0
        maxi=float("inf")

        for i in range(n):
            if prices[i]<maxi:
                maxi=prices[i]
            elif prices[i]-maxi>profit:
                profit=prices[i]-maxi

        
        return profit


        

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {

        long long mini=INT_MAX;
        int profit=0;
        for(auto num:prices){
            if(num<mini){
                mini=num;
            }

            else if (num-mini>profit){
                profit=num-mini;
            }

        }

        return profit;
    }
};